lib/Elastica/Aggregation/Filter.php

```diff
@@ -41,11 +41,11 @@ public function setFilter(AbstractFilter $filter)
     public function toArray()
     {
         $array = array(
-            'filter' => $this->getParam('filter'),
+            'filter' => $this->getParam('filter')->toArray(),
```

Comment:
I'm wrong, `getParam` will throw `InvalidException`. Still, it's better to check that was returned.